taichi/codegen/codegen_llvm.cpp

```diff
@@ -1464,11 +1502,18 @@ void CodeGenLLVM::store_floats_with_shared_exponents(BitStructStoreStmt *stmt) {
         sign_bit = builder->CreateLShr(sign_bit, 31 - cft->get_digit_bits());
         digits = builder->CreateOr(digits, sign_bit);
       }
-      store_custom_int(llvm_val[stmt->ptr],
-                       tlctx->get_constant(digits_bit_offset),
-                       cft->get_digits_type()->as<CustomIntType>(), digits);
+
+      // store the digits
+      val = builder->CreateZExt(
+          digits, llvm_type(bit_struct_physical_type->get_compute_type()));
```

Comment:
```suggestion
      val = builder->CreateZExt(digits, llvm_type(bit_struct_physical_type));
```
IIRC `bit_struct_physical_type` must be a primitive type, whose compute type would be itself.